src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp

```diff
@@ -798,8 +801,10 @@ ShenandoahScanRemembered<RememberedSet>::process_region(ShenandoahHeapRegion *re
   unsigned int cluster_size = CardTable::card_size_in_words * ShenandoahCardCluster<ShenandoahDirectCardMarkRememberedSet>::CardsPerCluster;
   size_t num_clusters = (size_t) ((num_heapwords - 1 + cluster_size) / cluster_size);
 
-  // Remembered set scanner
-  process_clusters(start_cluster_no, num_clusters, end_of_range, cl, use_write_table);
+  if (!region->is_humongous_continuation()) {
```

Comment:
Is this safe? why do we no longer need to scan humongous continuations?

Reply:
humongous continuations are scanned when the humongous start that "governs" the continuation is scanned.  I'll add a comment here.

Reply:
Wouldn't we need to change how `end_of_range` is computed for humongous starts?

Reply:
That's a good catch.  I'm a bit inconsistent with this.  I think this is "harmless" because end_of_range does not limit the scanning within a single humongous object.  It only limits the search for the "next object" following an initially scanned object.  I agree I should fix this up to make the code less confusing.  (Do you agree that the error is "harmless as is"?   Would appreciate your careful eyes on this.)

Reply:
`end_of_range` is used earlier to compute the `count` parameter (which is named `num_clusters` in the calling function). As I read it, this _does_ limit the number of cards scanned in the `process_clusters` call.

Reply:
The way I read it, if you call process_clusters with an address that represents the start of a humongous region, it will go through its object-processing loop at least once (p < endp first time through) and will completely scan the object found at p, which will be a humongous object that spans multiple following regions (which may exceed count clusters scanned in total and may reach reach well beyond endp in total scan).  Do you agree with this?

Reply:
Ah, okay - the `obj->oop_iterate` (or more likely, `array->oop_iterate_range`) will cover the entire humongous object if there is a dirty card in the start region. If the humongous object has no dirty cards in the start region, the last card for the object will be past the end card for this cluster and we'll `oop_iterate` it anyway.